### ISSUE TITLE
Consturuct randomly dying. 

### ISSUE THREAD

**Issue by tsukaerujuriki2** [Mithion/Mana-And-Artifice#921]
**Describe the bug**
I made a setup to reproduce and butcher animals, but at some point the construct just disappears and dies.

**Screenshots**

![2024-08-24_12 33 11](https://github.com/user-attachments/assets/16e9bc67-a148-4e0c-9e44-2ff845720ad1)
![2024-08-24_12 24 19](https://github.com/user-attachments/assets/d78ad75c-5ab3-4874-87f8-7f9b22c635be)
![2024-08-24_12 26 16](https://github.com/user-attachments/assets/38cf4c14-df9b-4618-a1d1-1ecd0aa69971)

**Version Information (please complete the following information):**
 - Mod Version: [3.0.0.22]
 - Forge Version [47.2.0]

**Additional context**

I am using construct with ender arms, and maybe this has something to do with it. Or, I got like 4 more constructs in the base so maybe there's too many?

I was playing on multiplayer server.

Add any other context about the problem here.  Other mods, etc.  For crash reports please use a service like https://pastebin.com/ or https://hastebin.com/.

**Comment by Mithion** [comment]
That death message is related to a construct dying to a damage source (not necessarily from an entity).

Do you have things like animals fighting back, fires, or other things it can wander into?  They should be immune to in_wall damage and things like that (although it can be datapacked out).

Constructs are tagged to not be removed, so it isn't despawning - it's dying to something.  Have you seen the death happen?

**Comment by tsukaerujuriki2** [comment]
No, Infact I didn't even see it get damaged once. The area it operated on is small so it's unlikely that I missed it.  Just the three animasl and the chest that you can see at the bottom of the picture. 

I was near the area when it happended, but didn't see any enemys too.

There's no animals fighting back either. I first thought It teleported to the wrong place so I used glow @e but didn't see it anywhere.

**Comment by tsukaerujuriki2** [comment]
I just tested it out. It just randomly dissapeared with a death message infront of my eyes few seconds after I allocated it to the loadstar lol

**Comment by tsukaerujuriki2** [comment]
https://youtu.be/ufuhOygEujU

**Comment by Mithion** [comment]
Do you have forge's remove erroring entities turned on?

**Comment by tsukaerujuriki2** [comment]
It is set to off

**Comment by tsukaerujuriki2** [comment]
My mod list:
https://pastebin.com/4wAwSpjm

In also checked the latest command done by construct, which is either butcher, ot take item from container.

**Comment by tsukaerujuriki2** [comment]
Ok, figured out the problem. The construct targets an null entity, then teleports to an far away location.

they're not actually dead.

![Screenshot 2024-08-28 110539](https://github.com/user-attachments/assets/65afdf60-7c91-4358-844b-5ed6254b2589)

**Comment by Mithion** [comment]
Good find!  It's wild how the constructs didn't crash there, I'm pretty happy with that level of recovery!

**Comment by Mithion** [comment]
Resolved in 3.1.0.0
